persistence-cassandra-4.0/src/main/java/io/stargate/db/cassandra/impl/CassandraPersistence.java

```diff
@@ -177,6 +180,7 @@ protected void initializePersistence(Config config) {
 
     interceptor.initialize();
     stargateHandler().register(interceptor);
+    stargateHandler().setAuthorizationService(this.authorizationService);
```

Comment:
nit: I just realized that since we have a direct call path into the handler, we do not really have to use `AtomicReference` as the param type, we could use `AuthorizationService` and the atomic thing will only exist in the handler for concurrent access... just looks nicer I guess :)

Reply:
I mean we'd call `CassandraPersistence.setAuth...` from the activators and it would pass the value to the handler?

Reply:
Except this is called before the AuthorizationService is populated. This init runs so that we can register the persistence bundle after which the auth-* bundles register.

Reply:
We call `cassandraDB.setAuthorizationService(ref)` from the activator(s)... I guess we could call `cassandraDB.setAuthorizationService(service)` when the service is resolved. This way we will not have to have a shared `AtomicReference` object. I suggested sharing `AtomicReference` before because I was not sure we could have a direct method call chain into the query handler.